.travis/build.py

```diff
@@ -295,15 +312,58 @@ def build_project_configs(app_path, config):
 					build_count += 1
 					if result["status"] != 0:
 						status = False
+						if expected_file is not None:
+							expected_different[app_path] = []
+							expected_status = get_expected_result(expected_file, app_path, board, bd_ver)
+							if not expected_status:
+								print "This application compile failed, but the expected result is pass"
+								expected_different[app_path].append([app_path, board, bd_ver, cur_core])
 
 					results.append(result)
-	return status, results, build_count
+	return status, results, build_count, expected_different
+
+def get_expected_result(expected_file, app_path, board, bd_ver):
+	result = None
+	filesuffix = os.path.splitext(expected_file)[1]
+	if filesuffix == ".json":
+		with open(expected_file, "r") as f:
+			expected_result = json.load(f)
+			if app_path in expected_result:
+				if board in expected_result[app_path]:
+					if bd_ver in expected_result[app_path][board]:
+						result = False
+					else:
+						result = True
+				else:
+					result = True
+			else:
+				result = True
```

Comment:
If app_path is not in expected_result, the example should be defined as PASS, so result should be False